relentless/ensemble.py

```diff
@@ -28,94 +73,96 @@ def __init__(self, types, T, P=None, V=None, mu={}, N={}, kB=1.0, conjugates=Non
             if t in N:
                 self._N[t] = N[t]
             if t in mu:
                 self._mu[t] = mu[t]
 
-        # conjugates can be specified (and assumed correct), or they can be deduced
-        if conjugates is not None:
-            self._conjugates = tuple(conjugates)
-        else:
-            if self.P is not None and self.V is not None:
-                raise ValueError('Both P and V cannot be set.')
-            for t in self.types:
-                if self.mu[t] is not None and self.N[t] is not None:
-                    raise ValueError('Both mu and N cannot be set for type {}.'.format(t))
-
-            # build the set of conjugate variables from the constructor
-            conjugates = []
-            if self.P is None:
-                conjugates.append('P')
-            if self.V is None:
-                conjugates.append('V')
-            for t in self.types:
-                if self.mu[t] is None:
-                    conjugates.append('mu_{}'.format(t))
-                if self.N[t] is None:
-                    conjugates.append('N_{}'.format(t))
-            self._conjugates = tuple(conjugates)
+        # conjugates are deduced from the specified parameters
+        if self.P is not None and self.V is not None:
+            raise ValueError('Both P and V cannot be set.')
+        for t in self.types:
+            if self.mu[t] is not None and self.N[t] is not None:
+                raise ValueError('Both mu and N cannot be set for type {}.'.format(t))
+
+        # build the set of conjugate variables from the constructor
+        conjugates = []
+        if self.P is None:
+            conjugates.append('P')
+        if self.V is None:
+            conjugates.append('V')
+        for t in self.types:
+            if self.mu[t] is None:
+                conjugates.append('mu_{}'.format(t))
+            if self.N[t] is None:
+                conjugates.append('N_{}'.format(t))
+        self._conjugates = tuple(conjugates)
 
     @property
     def beta(self):
+        """float: The inverse temperature/thermal energy."""
         return 1./(self.kB*self.T)
 
     @property
     def V(self):
+        """float or int: The volume of the system."""
         return self._V
 
     @V.setter
     def V(self, value):
-        if 'V' in self.conjugates:
-            self._V = value
-        else:
-            raise AttributeError('Volume is not a conjugate variable.')
+        self._V = value
 
     @property
     def P(self):
+        """float or int: The pressure of the system."""
         return self._P
 
     @P.setter
     def P(self, value):
-        if 'P' in self.conjugates:
-            self._P = value
-        else:
-            raise AttributeError('Pressure is not a conjugate variable.')
+        self._P = value
 
     @property
     def N(self):
+        """dict: The number of particles for each specified type."""
         return self._N.todict()
```

Comment:
Let's expose `self._N` directly.
```suggestion
        return self._N
```